src/diffusers/models/embeddings.py

```diff
@@ -545,11 +545,14 @@ def get_1d_rotary_pos_embed(
     assert dim % 2 == 0
 
     if isinstance(pos, int):
-        pos = np.arange(pos)
+        pos = torch.arange(pos)
```

Comment:
This should also be passed a `device` argument to allocate it on the GPU. If this isn't on the GPU, then neither will the following Tensors.